crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -3500,75 +3501,73 @@ impl<'a> MockEndpoint<'a, RoomRelationsEndpoint> {
-            )))
-            .respond_with(ResponseTemplate::new(404).set_body_json(json!({
-                "errcode": "M_NOT_FOUND",
-                "error": "Not found"
-            })));
-        MatrixMock { server: self.server, mock }
-    }
+/// A prebuilt mock for a `PUT
+/// /_matrix/client/v3/user/{userId}/account_data/io.element.recent_emoji`
+/// request, which updates the recently used emojis in the account data.
+#[cfg(feature = "experimental-element-recent-emojis")]
+pub struct UpdateRecentEmojisEndpoint {
+    pub(crate) request_body: Option<Vec<(String, UInt)>>,
 }
 
-/// A prebuilt mock for the update global account data endpoint.
-pub struct UpdateGlobalAccountDataEndpoint;
+#[cfg(feature = "experimental-element-recent-emojis")]
+impl UpdateRecentEmojisEndpoint {
+    /// Creates a new instance of the recent update recent emojis mock endpoint.
+    fn new() -> Self {
+        Self { request_body: None }
+    }
+}
 
-impl<'a> MockEndpoint<'a, UpdateGlobalAccountDataEndpoint> {
-    /// Returns a mock for a successful global account data event.
-    pub fn ok(self, user_id: &UserId, event_type: GlobalAccountDataEventType) -> MatrixMock<'a> {
-        let mock = self
-            .mock
-            .and(path_regex(format!(
-                r"^/_matrix/client/v3/user/{user_id}/account_data/{event_type}"
-            )))
-            .respond_with(ResponseTemplate::new(200).set_body_json(()));
-        MatrixMock { server: self.server, mock }
+#[cfg(feature = "experimental-element-recent-emojis")]
+impl<'a> MockEndpoint<'a, UpdateRecentEmojisEndpoint> {
+    /// Returns a mock that will check the body of the request, making sure its
+    /// contents match the provided list of emojis.
+    pub fn match_emojis_in_request_body(self, emojis: Vec<(String, UInt)>) -> Self {
+        Self::new(
+            self.server,
+            self.mock.and(body_json(json!(RecentEmojisContent::new(emojis)))),
+            self.endpoint,
+        )
     }
 
-    /// Returns a mock for a successful global account data event, as long as it
-    /// matches the provided `body`.
-    pub fn ok_with_request_body(
-        self,
-        user_id: &UserId,
-        event_type: GlobalAccountDataEventType,
-        body: Value,
-    ) -> MatrixMock<'a> {
-        let mock = self
-            .mock
-            .and(path_regex(format!(
-                r"^/_matrix/client/v3/user/{user_id}/account_data/{event_type}"
-            )))
-            .and(body_json(body))
-            .respond_with(ResponseTemplate::new(200).set_body_json(()));
+    /// Returns a mock for a successful update of the recent emojis account data
+    /// event. The request body contents should match the provided emoji
+    /// list.
+    #[cfg(feature = "experimental-element-recent-emojis")]
+    pub fn ok(self, user_id: &UserId) -> MatrixMock<'a> {
+        let mock =
+            global_account_data_mock_builder(self.mock, user_id, "io.element.recent_emoji".into())
```

Comment:
Can we use a `GlobalAccountDataEventType` variant instead of the custom variant here?

Reply:
Not unless we instantiate a dummy `RecentEmojisContent` instance so we can call `content.event_type()`, since the type is not part of ruma (yet) so that means it'll be a `GlobalAccountDataEventType::_Custom(here_be_dragons)` case.